src/login/connect_server.h

```diff
@@ -0,0 +1,2257 @@
+
+    if (sql->Query(Query, charid, accid, createchar->m_name, createchar->m_zone, createchar->m_nation) == SQL_ERROR)
+    {
+        ShowDebug(fmt::format("lobby_ccsave: char<{}>, accid: {}, charid: {}", createchar->m_name, accid, charid));
+        return -1;
+    }
+
+    Query = "INSERT INTO char_look(charid,face,race,size) VALUES(%u,%u,%u,%u);";
+
+    if (sql->Query(Query, charid, createchar->m_look.face, createchar->m_look.race, createchar->m_look.size) == SQL_ERROR)
+    {
+        ShowDebug(fmt::format("lobby_cLook: char<{}>, charid: {}", createchar->m_name, charid));
+        return -1;
+    }
+
+    Query = "INSERT INTO char_stats(charid,mjob) VALUES(%u,%u);";
+
+    if (sql->Query(Query, charid, createchar->m_mjob) == SQL_ERROR)
+    {
+        ShowDebug(fmt::format("lobby_cStats: charid: {}", charid));
+        return -1;
+    }
+
+    Query = "INSERT INTO char_exp(charid) VALUES(%u) \
+            ON DUPLICATE KEY UPDATE charid = charid;";
+    if (sql->Query(Query, charid, createchar->m_mjob) == SQL_ERROR)
+    {
+        return -1;
+    }
+
+    Query = "INSERT INTO char_jobs(charid) VALUES(%u) \
+            ON DUPLICATE KEY UPDATE charid = charid;";
+    if (sql->Query(Query, charid, createchar->m_mjob) == SQL_ERROR)
+    {
+        return -1;
+    }
+
+    Query = "INSERT INTO char_points(charid) VALUES(%u) \
+            ON DUPLICATE KEY UPDATE charid = charid;";
+    if (sql->Query(Query, charid, createchar->m_mjob) == SQL_ERROR)
+    {
+        return -1;
+    }
+
+    Query = "INSERT INTO char_unlocks(charid) VALUES(%u) \
+            ON DUPLICATE KEY UPDATE charid = charid;";
+    if (sql->Query(Query, charid, createchar->m_mjob) == SQL_ERROR)
+    {
+        return -1;
+    }
+
+    Query = "INSERT INTO char_profile(charid) VALUES(%u) \
+            ON DUPLICATE KEY UPDATE charid = charid;";
+    if (sql->Query(Query, charid, createchar->m_mjob) == SQL_ERROR)
+    {
+        return -1;
+    }
+
+    Query = "INSERT INTO char_storage(charid) VALUES(%u) \
+            ON DUPLICATE KEY UPDATE charid = charid;";
+    if (sql->Query(Query, charid, createchar->m_mjob) == SQL_ERROR)
+    {
+        return -1;
+    }
+
+    Query = "DELETE FROM char_inventory WHERE charid = %u";
+    if (sql->Query(Query, charid) == SQL_ERROR)
+    {
+        return -1;
+    }
+
+    Query = "INSERT INTO char_inventory(charid) VALUES(%u);";
+    if (sql->Query(Query, charid, createchar->m_mjob) == SQL_ERROR)
+    {
+        return -1;
+    }
+
+    if (settings::get<bool>("main.NEW_CHARACTER_CUTSCENE"))
+    {
+        Query = "INSERT INTO char_vars(charid, varname, value) VALUES(%u, '%s', %u);";
+        if (sql->Query(Query, charid, "HQuest[newCharacterCS]notSeen", 1) == SQL_ERROR)
+        {
+            return -1;
+        }
+    }
+    return 0;
+}
+
+int32 createCharacter(session_t& session, char* buf)
+{
+    auto sql = std::make_unique<SqlConnection>();
+    // Seed the random number generator.
+    srand(clock());
```

Comment:
Do we need this? Presumably a little later we're going to use xirand for something, so that fine?

Reply:
we're using xirand now